purchase_work_acceptance_invoice_plan/wizard/select_work_acceptance_invoice_plan_wizard.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SelectWorkAcceptanceInvoicePlanWizard(models.TransientModel):
+    _name = "select.work.acceptance.invoice.plan.wizard"
+    _description = "Select Work Acceptance Invoice Plan Wizard"
+
+    active_installment_ids = fields.Many2many(
+        comodel_name="purchase.invoice.plan",
+        compute="_compute_active_invoice_plan_ids",
+    )
+    installment_id = fields.Many2one(
+        comodel_name="purchase.invoice.plan",
+        string="Invoice Plan",
+        required=True,
+        domain="[('id', 'in', active_installment_ids)]",
+    )
+
+    @api.depends("installment_id")
+    def _compute_active_invoice_plan_ids(self):
+        self.ensure_one()
+        purchase = self.env["purchase.order"].browse(
+            self._context.get("active_ids", [])
+        )
+        installment_ids = (
+            purchase.wa_ids.filtered(lambda l: l.state == "accept")
+            .mapped("installment_id")
+            .ids
+        )
+        self.active_installment_ids = self.env["purchase.invoice.plan"].search(
+            [("purchase_id", "=", purchase.id), ("id", "not in", installment_ids)]
+        )
+
+    def button_create_wa(self):
+        purchase = self.env["purchase.order"].browse(self._context.get("active_id"))
```

Comment:
Let's use `self.ensure_one()` for this function. Because it is not possible to have multiple record.